backend/services/mgmt/v1alpha1/connection-service/connection.go

```diff
@@ -163,6 +163,51 @@ func (s *Service) CheckConnectionConfig(
 	}
 }
 
+func (s *Service) CheckConnectionConfigById(
+	ctx context.Context,
+	req *connect.Request[mgmtv1alpha1.CheckConnectionConfigByIdRequest],
+) (*connect.Response[mgmtv1alpha1.CheckConnectionConfigByIdResponse], error) {
+	connectionUuid, err := nucleusdb.ToUuid(req.Msg.Id)
+	if err != nil {
+		return nil, err
+	}
+
+	conn, err := s.db.Q.GetConnectionById(ctx, s.db.Db, connectionUuid)
+	if err != nil && !nucleusdb.IsNoRows(err) {
+		return nil, err
+	} else if err != nil && nucleusdb.IsNoRows(err) {
+		return nil, nucleuserrors.NewNotFound("unable to find connection by id")
+	}
+
+	_, err = s.verifyUserInAccount(ctx, nucleusdb.UUIDString(conn.AccountID))
+	if err != nil {
+		return nil, err
+	}
+
+	dtoVal, err := dtomaps.ToConnectionDto(&conn)
+	if err != nil {
+		return nil, err
+	}
+
+	checkReq := &connect.Request[mgmtv1alpha1.CheckConnectionConfigRequest]{
+		Msg: &mgmtv1alpha1.CheckConnectionConfigRequest{
+			ConnectionConfig: dtoVal.ConnectionConfig,
+		},
+	}
+
+	resp, err := s.CheckConnectionConfig(ctx, checkReq)
+	if err != nil {
+		return nil, err
+	}
+
+	return connect.NewResponse(&mgmtv1alpha1.CheckConnectionConfigByIdResponse{
+		IsConnected:     resp.Msg.IsConnected,
+		ConnectionError: resp.Msg.ConnectionError,
+		Privileges:      resp.Msg.Privileges,
```

Comment:
It's good practice to use the DTO Getters instead of accessing the properties directly, at least when the props are non-nil.
```suggestion
		IsConnected:     resp.Msg.GetIsConnected(),
		ConnectionError: resp.Msg.GetConnectionError(), // maybe don't need this one if it is a pointer and go is unhappy with the getter
		Privileges:      resp.Msg.GetPrivileges(),
```